api/tests/test_asset_log.py

```diff
@@ -337,6 +338,109 @@ def test_authenticated_admin_user_get_of_asset_logs_invalid_filter(
         self.assertEqual(response.status_code, 200)
         self.assertEqual(len(response.data["results"]), 0)
 
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_year(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = (
+            f"{self.asset_logs_url}?year={self.test_other_asset.created_at.year}"
+        )
+
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        data = response.data["results"]
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            data[0]["created_at"].split("-")[0],
+            f"{self.test_other_asset.created_at.year}",
+        )
```

Comment:
Let your assertion show the number of items filtered by the query. You can do the same for other assertions.

Reply:
@kmazi Wouldn't this mean that I first have to show the number of items that are there before the filter?
All the dates are autogenerated and get (datetime.now()) each time you run the test
I tried updating the `created_at` time and it would resolve to a zero everytime.